src/NzbDrone.Core/Download/Clients/Porla/README.md

```diff
@@ -0,0 +1,63 @@
+# Porla Client
+
+[Porla](https://porla.org/)! The (Unofficial) Web frontend for [LibTorrent](https://libtorrent.org/)!
+
+## Details
+
+* Uses JSONRPC
+* Written in C++
+* Has LUA Plugin Capability
+* Session Capable
+* Has Proxy Support (Looking at you Transmission!)
+* Presets!
+
+> Currently (2024-02) still in bleeding-edge active development.
+
+## Deving
+
+I coppied large parts of this code from the Hadouken Client, as it was the other JSONRPC client in the collection.
+
+Docs are iffy currently, check out the methods at the [source (v0.37.0)](https://github.com/porla/porla/tree/v0.37.0/src/methods). They are not too hard to figure out in conjunction with the [method docs](https://porla.org/api/methods/)
+
+### `GetStatus()`
+
+Unsure what this needs to do exactly.  
+I _think_ it's supposed to introspect the configuration.  
+_IF_ that is the case. TODO: Figure out extracting the `save_path` from the prefix 
+
+## Feature Implementation
+
+### Sessions
+
+I **DID NOT** do anything with sessions.
+
+They could be VERY powerfull to work with Private and Public trackers, but the front end currently doesn't have strong support for it.
+
+Could also be a way to have multiple Sonarrs pointed at the same Porla, or a diffrent way to deal with the other Starr ecosystem.
+
+### Presets
+
+Presets are being set on the torrent adds
+
+So I needed to do something with the `save_path`.  
+I wanted this to work "by default", so if you have NO presets set, you **NEED** to set the `save_path` _BUT_ if you want to use the presets values you shoule NOT set the `save_path` in the RPC request. That is why you see that intersting branch on the `AddTorrent` functions.
+
+### Tags
+
+Would be great to use to filter BIG lists of torrents even more! But it doesn't look like Sonarr expects that. looks like it expects everything
```

Comment:
Sonarr used tags/categories/labels in the download client to reduce the number of items it gets from the client (or if not filtered on the client, filtered while processing all results), they aren't used to filter by series or anything more specific.

Reply:
But to get that list Sonarr asks for everything from the client.
It's not "On-Demand" correct?

So I we have say 1000 active torrents in the client, everything is requested, and only the ones that make sense to sonarr are loaded into the db, and THEN filtered because Sonarr can do the Series/Episode/Category mapping inside itself.

We have the capability to do optimized queries to porla: "Give us all the torrents with the tag `TVDBID=123456789` in the catagory `sonarr-tv` please.".

Of course in, practice that isn't too big of a concern really.
I still added the tags because I would very much like to filter the items outside of the Sonarr's context (dealing with said 1000 actives).